src/createSignal.spec.lua

```diff
@@ -86,4 +86,69 @@ return function()
 		-- Exactly once listener should have been called.
 		expect(spyA.callCount + spyB.callCount).to.equal(1)
 	end)
+
+	it("should allow adding listener in the middle of firing", function()
+		local signal = createSignal()
+
+		local disconnectA
+		local spyA = createSpy()
+		local listener = function(a, b)
+			disconnectA = signal:subscribe(spyA.value)
+		end
+
+		local disconnectListener = signal:subscribe(listener)
+
+		expect(spyA.callCount).to.equal(0)
+
+		local a = {}
+		local b = 67
+		signal:fire(a, b)
+
+		expect(spyA.callCount).to.equal(0)
+
+		-- The new listener should be picked up in next fire.
+		signal:fire(b, a)
+		expect(spyA.callCount).to.equal(1)
+		spyA:assertCalledWith(b, a)
+
+		disconnectA()
+		disconnectListener()
+
+		signal:fire(a)
+
+		expect(spyA.callCount).to.equal(1)
+	end)
+
+	it("should have one connection instance when add the same listener multiple times", function()
```

Comment:
For what it's worth, we try to be cautious about warnings when Roact powers things like in-game menu and studio plugins. React doesn't actually expose its signal implementation, so warning about this would signal an _internal_ bug. If this cropped up in shipped code, we'd end up confusingly printing warnings to developers in studio about things that have nothing to do with them!

Reply:
I've added a warning message.

Reply:
Yeah I'm a bit worried about the warning too, I am not sure if that can bubble up to the consumers of Roact. I've checked out createContext and Binding and it seems ok to me

Reply:
I will remove it for now. We can loop back on the warning message in a separate PR if necessary